### PR TITLE
externalContracts section tweaks

### PR DESCRIPTION
**Pull request by technophile-04**
### Description: 

- Moved the externalContracts section to its own page.
- Tweaked `hooks.md` page with some more context around `deployedContracts.ts`
- Added a note in ` useScaffoldWatchContractEvent` about https://github.com/scaffold-eth/scaffold-eth-2/issues/838

### CODE REVIEW

File: docs/external-contracts/external-contracts.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+sidebar_position: 5
+---
+
+# 🛠 Interacting with External Contracts
```
Comment (carletex) [inline docs/external-contracts/external-contracts.md:5]: How about any of these? 🌐 📡 🔗

**Comment by carletex** [comment]
Love this @technophile-04 Thanks! It was hard to find before. Just left a emoji comment :D

File: docs/external-contracts/external-contracts.md (hunk 1 above)
Comment (technophile-04) [reply docs/external-contracts/external-contracts.md]: Ohh yeah nice!, used 📡  at https://github.com/scaffold-eth/se2-docs/pull/74/commits/434b53a12dbf0818b04c54ed6e6429951cdb36e8

**Comment by technophile-04** [comment]
Merging this 🙌

**Pull request merged**